webapp/src/js/controllers/messages-content.js

```diff
@@ -115,9 +115,35 @@ angular.module('inboxControllers').controller('MessagesContentCtrl',
         getContactable(id, type),
         MessageContacts.conversation(id)
       ])
+        .catch(function(err) {
```

Comment:
Because you've put this catch against the `$q.all`, in situations where you get a 404 you're going to call `MessageContacts.conversation` twice. 

To avoid this, move this catch into an earlier part of the code flow: I would recommend in the `getContactable` function itself, against the call that could cause the error to occur. This has the added benefit of more clearly associating this edge case with the code that causes it.